index.js

```diff
@@ -204,7 +213,16 @@ function appInit (opts) {
   // enable catching <href a=""></href> links
   // enable HTML5 history API
   if (opts.href !== false) pushLocationSub(href)
-  if (opts.history !== false) pushLocationSub(history)
+  if (opts.history !== false) {
+    pushLocationSub(history)
+  // otherwise enable hash history
+  } else {
+    pushLocationSub(function (navigate) {
```

Comment:
The only way I can think to shorten this like the other two is to use function composition, which would require a `compose` function or importing one from a library.